### ISSUE TITLE
Improve consistency of classification loss functions across curvatures (and support both sigmoid and sofmax cross entropy loss)

### ISSUE THREAD

**Issue by Thomas-Christie** [laplax-org/laplax#27]
As discussed in person, currently there are different expectations on the format of the model's output/task setup when using `ggn` vs `hessian` for curvature in the case of classification tasks:
- `ggn` assumes the model outputs `C` per-class logits for the case of `C`-class classification, resulting in the standard cross-entropy loss using `softmax` applied to the logits produced by the model (used e.g. [here](https://github.com/laplax-org/laplax/blob/main/laplax/curv/ggn.py#L29)).
- `hessian` assumes the model is performing binary classification, with the model outputting a *single* logit, which is transformed using the *sigmoid* function. As such, the resulting loss used is a [log_sigmoid_cross_entropy](https://github.com/laplax-org/laplax/blob/main/laplax/curv/hessian.py#L23).

As such, the aims of this issue are twofold:
1. In the case of binary classification, enable the user to specify whether their model is outputting *two* logits (in which case we use the `softmax` activation function in the loss) or a *single* logit (in which case we use the `sigmoid` activation in the loss).
2. Provide a unified interface for specifying the loss setup across the different curvature techniques in the case of classification problems (i.e. `multiclass classification`, `binary classification with single output` and `binary classification with two outputs`).
